md2cf/api.py

```diff
@@ -256,3 +267,86 @@ def get_space(self, space, additional_expansions=None):
+        additional_expansions=None,
+    ):
+        """
+        Gets content descendant
+
+        Args:
+            title (str): the title for the page
+            space_key (str): the Confluence space for the page
+            content_type (str): Content type. Default value: page.
+              Valid values: page, blogpost.
+            page_id (str or int): the ID of the page
+            additional_expansions (list of str): Additional expansions that should be
+              made when calling the api
+
+        Returns:
+            The response from the API
+
+        """
+        params = None
+        if additional_expansions is not None:
+            params = {"expand": ",".join(additional_expansions)}
+
+        if page_id is not None:
+            response = self._get(
+                f"content/{page_id}/descendant/{content_type}", params=params)
+            results = response.results
+            while (hasattr(response, '_links') and hasattr(response._links, 'next')):
+                response = self._get(response._links.next.replace(
+                    '/rest/api/', ''), params=params)
+                results.extend(response.results)
+            return results
+        elif title is not None:
+            params = {"title": title, "type": content_type}
+            if space_key is not None:
+                params["spaceKey"] = space_key
+            response = self._get("content", params=params)
+            try:
+                # A search by title/space doesn't return full page objects,
+                # and since we don't support expansion in this implementation
+                # just yet, we just retrieve the "full" page data using the page
+                # ID for the first search result
+                return self.get_content_descendant(
+                    page_id=response.results[0].id,
+                    additional_expansions=additional_expansions
+                )
+            except IndexError:
+                return None
+        elif space_key is not None:
+            space_homepage_id = self._get(
+                f"space/{space_key}")._expandable.homepage.replace('/rest/api/content/', '')
+            return self.get_content_descendant(page_id=space_homepage_id)
+        else:
+            raise ValueError(
+                "At least one of title or page_id or space_key must not be None")
+
+    def purge_page(
+        self,
+        page=None
+    ):
+        """
+        Delete page in a space
+
+        Args:
+            page (page): the page to be purged
+
+        Returns:
+            The response from the API
+
+        """
+        if page is not None:
+            params = {"status": "trashed"}
+            page_from_get = self.get_page(page_id=page.id)
+            self.update_page(page=page_from_get, body="", status="trashed")
+            return self._delete(f"content/{page.id}", params=params)
+        else:
+            raise ValueError("Page cannot be None")
```

Comment:
Since the parameter is not optional, I would just remove the `=None` and the error handling :)

```suggestion
    def purge_page(
        self,
        page
    ):
        """
        Delete page in a space

        Args:
            page (page): the page to be purged

        Returns:
            The response from the API

        """
        params = {"status": "trashed"}
        page_from_get = self.get_page(page_id=page.id)
        self.update_page(page=page_from_get, body="", status="trashed")
        return self._delete(f"content/{page.id}", params=params)
```

Reply:
Less code is better indeed!